src/hooks/useAccounts.ts

```diff
@@ -1,42 +1,16 @@
-import axios, { AxiosResponse } from 'axios';
-import { useCallback, useEffect, useState } from 'react';
-import { withAxios } from '../services/axios';
-import { AccountsResponse, BaseHook } from '../types';
-
-interface AccountsService extends BaseHook {
-    data: {
-        accounts: AccountsResponse['data'];
-    };
-}
+import { accountsEndpoint, AccountsResponse } from 'endpoints/accounts';
+import { useAsync } from 'hooks/useAsync';
+import { useEffect } from 'react';
 
 function useAccounts() {
-    const [accounts, setAccounts] = useState<
-        AccountsService['data']['accounts']
-    >([]);
-    const [error, setError] = useState<string | null>(null);
-    const [loading, setLoading] = useState<boolean>(true);
-
-    const fetchConnectors = useCallback(async () => {
-        setLoading(true);
-        setError(null);
-        withAxios(axios.get('/accounts'), setError, setLoading)
-            .then((response: AxiosResponse<AccountsResponse>) => {
-                setAccounts(response.data.data);
-            })
-            .catch(() => {});
-    }, []);
+    const response = useAsync<AccountsResponse>();
+    const { run } = response;
```

Comment:
We no no longer need each hook to handle their own state of loading, errors, data, etc. We just pass the useAsync hook right now. This could be bad... like we might want to make a mediator type and only return certain things. Like we don't care much about 'idle' yet so there is some extra fluff being passed along. I think as we work with this and see how it feels to use we'll figure out a good pattern